src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged/DeflateManagedStream.cs

```diff
@@ -136,7 +139,13 @@ public override int Read(byte[] buffer, int offset, int count)
                 _inflater.SetInput(_buffer, 0, bytes);
             }
 
-            return count - remainingCount;
+            return initialLength - buffer.Length;
+        }
+
+        public override int ReadByte()
+        {
+            byte b = default;
+            return Read(MemoryMarshal.CreateSpan(ref b, 1)) == 1 ? b : -1;
```

Comment:
Nice, I was not aware of the existence of `MemoryMarshal.CreateSpan`. We should most probably use it in `FileStream` as well:

https://github.com/dotnet/runtime/blob/419506b13f45ebfd4d5459accde27a20c5482f7c/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/WindowsFileStreamStrategy.cs#L148-L150